packages/react-components/src/components/DomainPermissions.tsx

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import React from 'react';
+import { Stack, Text, Link, Icon } from '@fluentui/react';
+/* @conditional-compile-remove(call-readiness) */
+import { useLocale } from '../localization';
+/* @conditional-compile-remove(call-readiness) */
+import { _formatString, _pxToRem } from '@internal/acs-ui-common';
+import {
+  iconBackDropStyles,
+  iconContainerStyles,
+  iconPrimaryStyles,
+  linkTextStyles,
+  primaryTextStyles,
+  secondaryTextStyles,
+  textContainerStyles
+} from './styles/DomainPermissions.styles';
+
+/**
+ * @internal
+ * Props for DomainPermissions component.
+ */
+export interface DomainPermissionsProps {
+  appName: string;
+  onGetTroubleShooting: () => void;
+  strings: DomainPermissionsStrings;
+}
+
+/**
+ * @beta
+ * Strings for DomainPermissions component
+ */
+export interface DomainPermissionsStrings {
+  mainText: string;
+  secondaryText: string;
+  linkText: string;
+}
+
+const _DomainPermissionsContainer = (props: DomainPermissionsProps): JSX.Element => {
+  const { appName, onGetTroubleShooting, strings } = props;
+
+  const appNameTrampoline = (): string => {
+    /* @conditional-compile-remove(call-readiness) */
```

Comment:
This whole component is internal, can we simply not conditionally compile any of it? Or otherwise conditionally compile the export in the index.ts file?

Reply:
will reapproach this, I do feel like there is a better way to do this but similar things are done in other beta components to make sure that it builds appropriately.

Reply:
Is there a way to have a top level conditional compile instead of many throughout?